dataikuapi/dss/ml.py

```diff
@@ -1047,7 +1057,10 @@ def __setattr__(self, attr_name, value):
                 elif isinstance(target, CategoricalHyperparameterSettings):
                     target.set_values(value)
                 elif isinstance(target, NumericalHyperparameterSettings):
```

Comment:
Maybe overthinking this, but would be interesting to be able to pass an instance of `NumericalHyperparameterSettings.Range`. I can imagine copying parameters from one algorithm to another for a particular hyperparameter.

Reply:
Hum. I didn't think of that but that makes sense !

Reply:
Actually, if the user can do that for `range`, we will want to allow a similar behaviour with `values`. So we need to be careful not to share just pointers and copy the inputs and document the behaviour.

Reply:
What about adding a `copy` (or `to_plain`)  method to `NumericalHyperparameterSettings.Range` that outputs a .... `Range` 😅 , so that we can copy from one algo to the other using the same syntactic sugar ?

This would not work for explicit values though, unless we create a `NumericalHyperparameterSettings.Values` or something...

Copying parameters between one algo to another using syntactic sugar requires a bit more thought to be consistent all around, IMO

Reply:
I guess it means arbitration between
- either holding a pointer to a state and being able to do `algo1.hp1.range.min = 3` that edits the parent ~dict
- or copying `.range` to another hp/algo/mltask?

Or maybe when calling the setter with a Range value, changing the pointer?

Reply:
We could do both, by adding a specific syntactic sugar for `NumericalHyperparameterSettings.Range` and copying the `min`, `max` and `nb_values` into `set_range`'s arguments, as Nicolas suggested (i.e drop the pointer). I'm not sure whether this is less confusing than adding a kind of `copy` method to `NumericalHyperparameterSettings.Range` as I mentionned above.

For copying `values` between algorithms, after thinking about it again, it seems that it would work out of the box, because calling a `NumericalHyperparameterSettings.values` property returns a `list`

Reply:
To simplify things a bit, what about being able to copy the _entire hyperparameter setting_ (except `scaling` and `limits` of course) when doing `algo1.hyperparameter1 = algo2.hyperparameter2` and not allowing any other form of copy ?

Reply:
IMO if we manage to create a consistent copy mechanism for the hyperparameter setting fields, then copying the entire setting would be trivial (looping through the fields and copying them one by one). But that would be another syntactic sugar that checks if `value` has the same type as `target` and performs the copy.

For the syntactic sugar that only copies the `NumericalHyperparameterSettings.Range`, I like Nicolas solution that drops the pointer and only copies the `min`, `max` and `nb_values` into `set_range`'s arguments

Reply:
I don't see a situation where the user would only want to copy either a `.range` or a `.values` field from one hyperparameter and not the whole thing. Because they will want to set it either as a `RANGE` or as `EXPLICIT` in which case they only care about either `.range` or `.values`. 

So allowing to copy either the whole hyperparameter setting (`algo1.hyperparameter1 = algo2.hyperparameter2`) or nothing, i.e. forbidding `algo1.hyperparameter1 = algo2.hyperparameter2.range` and `algo1.hyperparameter1 = algo2.hyperparameter2.values` sounds both clearer in terms of API (notice the type consistency at assignment) and simpler in terms of implementation.